### PR TITLE
fix(plonky2): fix prove verifier.toml output

### PR DESCRIPTION
**Pull request by Aristotelis2002**
Fixed the bug of mismatching data in the generated Verifier.toml file when using `nargo prove`. This was caused by an incorrect matching of the input data with the values in WitnessMap. More specifically the bug was caused by calling the `decode` method on the object returned by the function `public_abi` which filters needed parameters for the correct decoding process.

The fix consists of the change of not calling the `public_abi` before `decode` and passing a public_abi object to `write_inputs_to_file` which only writes the parameters in the given abi object into Verifier.toml. All those parameters are public.

Moved the bug related test into the expected success folder.

### CODE REVIEW

File: tooling/nargo_cli/src/cli/prove_cmd.rs [hunk 1]
```diff
@@ -125,11 +125,12 @@ pub(crate) fn prove_package(
     // Get the entry point witness for the ABI
     let main_witness =
         &witness_stack.peek().expect("Should have at least one witness on the stack").witness;
-    let public_abi = compiled_program.abi.public_abi();
-    let (public_inputs, return_value) = public_abi.decode(main_witness)?;
+    let abi = compiled_program.abi;
+    let (inputs, return_value) = abi.decode(main_witness)?;
```
Comment (smanilov) [inline tooling/nargo_cli/src/cli/prove_cmd.rs:129]: Naming: `all_inputs` might have been even better, but not a blocker.

File: test_programs/plonky2_verify_success/non_first_public_input/VerifierTest.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # This file is separate from Verifier.toml, because nargo prove overwrites Verifier.toml
-# Note: This test should fail, because this y takes the value of x, provided in Prover.toml,
-#  rather than the value of y in Prover.toml 
+# Note: This test should pass, because this y takes the value of y, provided in Prover.toml,
```
Comment (smanilov) [inline test_programs/plonky2_verify_success/non_first_public_input/VerifierTest.toml:2]: I was going to suggest that we drop the comment here, but on a second thought it's fine. Keep it.

**Review by smanilov — APPROVED** [review]
I ran the test locally and it passes.

Thanks for the good fix!

File: tooling/nargo_cli/src/cli/prove_cmd.rs (hunk 1 above)
Comment (Aristotelis2002) [reply tooling/nargo_cli/src/cli/prove_cmd.rs]: Done.

**Pull request merged**